redfish-core/lib/log_services.hpp

```diff
@@ -601,16 +616,25 @@ inline void
 
             if (dumpType == "BMC")
             {
+                thisEntry["@odata.id"] = entriesPath + entryID;
+                thisEntry["Id"] = entryID;
                 thisEntry["DiagnosticDataType"] = "Manager";
                 thisEntry["AdditionalDataURI"] = entriesPath + entryID +
                                                  "/attachment";
                 thisEntry["AdditionalDataSizeBytes"] = size;
             }
-            else if (dumpType == "System")
+
+            else if (dumpType == "System" || dumpType == "Resource" ||
+                     dumpType == "Hardware" || dumpType == "Hostboot" ||
+                     dumpType == "SBE")
             {
+                std::string dumpEntryId(dumpType + "_");
+                dumpEntryId.append(entryID);
+                thisEntry["@odata.id"] = entriesPath + dumpEntryId;
+                thisEntry["Id"] = dumpEntryId;
                 thisEntry["DiagnosticDataType"] = "OEM";
                 thisEntry["OEMDiagnosticDataType"] = "System";
-                thisEntry["AdditionalDataURI"] = entriesPath + entryID +
+                thisEntry["AdditionalDataURI"] = entriesPath + dumpEntryId +
```

Comment:
Would this is to be upstreamed?
This seems becoming different from upstream for `dumpType=="System"`?
If it is not going to be upstreamed, should this be separate from "System" dump -- and keep "System" dump as is?

Reply:
Actually all the host dumps are listed under Systems/system by Redfish
So here in bmcweb, we read different host dumps from the dbus and list it under "System" type. dumpEntryID is just to different between each of the host dumps.

And this is downstream only as of now because Dhruv's dump design proposal is targeted for pFuture.

Reply:
@asmithakarun 
My comment is more on the code difference for "System" between upstream and 1110.
If it will be handled in the future, I'm good on it.

Reply:
yes, pFuture will have all the dump types upstreamed, meaning these code differences will not be there